File: docs/src/testers/anything-api/start_detached.ts
// @sniptest filename=start_detached.ts
// @sniptest show=1-19
const NOTTE_API_KEY = process.env.NOTTE_API_KEY as string;

const response = await fetch('https://anything.notte.cc/api/anything/start', {
  method: 'POST',
  headers: {
    Authorization: `Bearer ${NOTTE_API_KEY}`,
    'Content-Type': 'application/json',
  },
  body: JSON.stringify({ query: 'fetch the top 3 hacker news posts', detach: true }),
});
if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

// 202 Accepted
const run = (await response.json()) as { thread_id: string; status: string; url: string };
console.log('Thread ID:', run.thread_id);
console.log('Status:', run.status); // "started"
console.log('Follow along at:', run.url);

export {};

File: docs/src/testers/anything-api/start_detached.py
# @sniptest filename=start_detached.py
# @sniptest typecheck_only=true
import os

import requests

NOTTE_API_KEY = os.environ["NOTTE_API_KEY"]

response = requests.post(
    "https://anything.notte.cc/api/anything/start",
    headers={
        "Authorization": f"Bearer {NOTTE_API_KEY}",
        "Content-Type": "application/json",
    },
    json={"query": "fetch the top 3 hacker news posts", "detach": True},
    timeout=30,
)
response.raise_for_status()  # 202 Accepted

run = response.json()
print("Thread ID:", run["thread_id"])
print("Status:", run["status"])  # "started"
print("Follow along at:", run["url"])
